src/main/java/eu/dissco/core/translator/terms/utils/AgentsUtils.java

```diff
@@ -0,0 +1,93 @@
+package eu.dissco.core.translator.terms.utils;
+
+import eu.dissco.core.translator.domain.AgenRoleType;
+import eu.dissco.core.translator.schema.Agent.Type;
+import eu.dissco.core.translator.schema.Agent;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+public class AgentsUtils {
+
+  private AgentsUtils() {
+    // This is a Utility class
+  }
+
+  public static void setAgent(List<Agent> currentAgents, String termValue, String id,
+      AgenRoleType role, eu.dissco.core.translator.schema.Agent.Type type) {
+    if (currentAgents == null){
+      currentAgents = new ArrayList<>();
+    }
+    if (termValue != null || id != null) {
+      if ((termValue != null && (termValue.contains("&") || termValue.contains("|"))) || (
+          id != null && (id.contains("&") || id.contains("|")))) {
+        handleMultipleAgents(currentAgents, termValue, id, role, type);
+      } else {
+        constructAgent(currentAgents, termValue, id, role, type);
+      }
+    }
+  }
+
+  private static void constructAgent(List<Agent> currentAgents, String termValue, String id,
+      AgenRoleType role, Type type) {
+    String name = termValue;
+    if (termValue != null && termValue.contains("http") && id == null) {
+      id = termValue;
+      name = null;
+    } else if (termValue != null && termValue.contains("http") && id != null
+        && !id.equals(termValue)) {
+      log.warn(
+          "The id and term value do not match for term: {} and id: {}. Adding value as name and id as identifier",
+          termValue, id);
+    }
+    var agent = new Agent()
+        .withId(id)
+        .withType(type)
+        .withSchemaName(name)
+        .withSchemaIdentifier(id)
+        .withOdsHasRoles(
+            List.of(new eu.dissco.core.translator.schema.OdsHasRole().withType("schema:Role")
+                .withSchemaRoleName(role.getName())));
+    if (id != null) {
+      agent.withOdsHasIdentifiers(List.of(
+          new eu.dissco.core.translator.schema.Identifier().withId(id)
+              .withType("ods:Identifier")
+              .withDctermsIdentifier(id)));
+    }
+    currentAgents.add(agent);
+  }
+
+  private static void handleMultipleAgents(
+      List<Agent> currentAgents, String termValue, String id, AgenRoleType role,
+      eu.dissco.core.translator.schema.Agent.Type type) {
+    var ids = new String[0];
+    var agents = new String[0];
```

Comment:
maybe call this `agentNames` to distinguish it from the Agent object

Reply:
That would imply that this would contain names but it could actually contain ids. So agents keep it generic, later we will check if this is a resolvable identifier or not